File: src/backend/log.ts
import { invoke } from "@tauri-apps/api/tauri";

import type { Encoding } from "../types/encode";

export async function log(message: String) {
    return invoke("on_log" , {
        message: message,
    });
}

export async function logError(context: string, error: Error) {
    return invoke("on_log_error", {
        context: context,
        message: error.message + "\n\n" + error.stack,
      });
}

File: src-tauri/src/log.rs
use colored::*;

pub fn log(_message: &str) {
    // println!("{}", message);
    // println!();
}

pub fn log_lines(lines: Vec<&str>) {
    for line in lines {
        log(line);
    }
}

pub fn log_error(context: &str, message: &str) {
    println!("{} {}", "Error:".bright_red(), context.bright_red());
    println!("       {}", message);
    println!();
}
